packages/mdc-tab/index.ts

```diff
@@ -0,0 +1,143 @@
+/**
+ * @license
+ * Copyright 2018 Google Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+import {MDCComponent} from '@material/base/component';
+import {MDCRipple, MDCRippleFoundation, RippleCapableSurface} from '@material/ripple/index';
+import {MDCTabIndicator} from '@material/tab-indicator/index';
+import {MDCTabFoundation} from './foundation';
+import {MDCTabDimensions, RippleFactory, TabIndicatorFactory, TabInteractionEventDetail} from './types';
+
+class MDCTab extends MDCComponent<MDCTabFoundation> implements RippleCapableSurface {
+  static attachTo(root: Element): MDCTab {
+    return new MDCTab(root);
+  }
+
+  id!: string; // assigned in initialize();
+
+  // Public visibility for this property is required by RippleCapableSurface.
+  root_!: HTMLElement; // assigned in MDCComponent constructor
+
+  private ripple_!: MDCRipple; // assigned in initialize();
+  private tabIndicator_!: MDCTabIndicator; // assigned in initialize();
+  private content_!: HTMLElement; // assigned in initialize();
+  private handleClick_!: EventListener; // assigned in initialize();
```

Comment:
typed to SpecificEventListener<'click'>

also assigned in initialSyncWithDom

Reply:
Done